crates/sui-core/src/node_sync/node_state.rs

```diff
@@ -370,46 +425,55 @@ where
                 .await
                 .map_err(|_| SuiError::TimeoutError);
 
-                let digest = sync_arg.transaction_digest();
+                let tx_digest = sync_arg.transaction_digest();
 
                 let res = match res {
                     Err(error) | Ok(Err(error)) => {
-                        error!(?digest, "process_digest failed: {}", error);
+                        error!(?tx_digest, "process_digest failed: {}", error);
                         Err(error)
                     }
 
                     Ok(Ok(res)) => {
                         // Garbage collect data for this tx.
                         if let SyncStatus::CertExecuted = res {
-                            state.cleanup_cert(epoch_id, digest);
+                            state.cleanup_cert(epoch_id, tx_digest);
                         }
                         Ok(res)
                     }
                 };
 
                 // Notify waiters even if tx failed, to avoid leaking resources.
-                trace!(?epoch_id, ?digest, "notifying parents and waiters");
-                Self::notify(&state.pending_parents, digest, res.clone()).await;
-                Self::notify(&state.pending_txes, digest, res.clone()).await;
-
-                if let Some(tx) = tx {
-                    // Send status back to follower so that it knows whether to advance
-                    // the watermark.
-                    if tx.send(res).is_err() {
-                        // This will happen any time the follower times out and restarts, but
-                        // that's ok - the follower won't have marked this digest as processed so it
-                        // will be retried.
-                        debug!(
-                            ?sync_arg,
-                            ?epoch_id,
-                            "could not send process_digest response to caller",
-                        );
-                    }
-                }
+                trace!(?epoch_id, ?tx_digest, "notifying parents and waiters");
+                Self::notify(&state.pending_parents, tx_digest, res.clone()).await;
+                Self::notify(&state.pending_txes, tx_digest, res.clone()).await;
+
+                Self::send_sync_res_to_receiver(tx, res, &sync_arg, &epoch_id);
             });
         }
     }
 
+    fn send_sync_res_to_receiver(
+        tx: Option<oneshot::Sender<Result<SyncStatus, SuiError>>>,
+        res: Result<SyncStatus, SuiError>,
+        sync_arg: &SyncArg,
+        epoch_id: &EpochId,
+    ) {
+        if let Some(tx) = tx {
+            // Send status back to follower so that it knows whether to advance
+            // the watermark.
```

Comment:
fwiw this comment is obsolete, there's no water mark anymore. I'm not sure any comment is necessary here at all.